src/plugins/saved_search/common/constants.ts

```diff
@@ -8,6 +8,10 @@
  */
 
 export const SavedSearchType = 'search';
+// While the legacy SO name has to stay "search" the display name can be customized.
+// We can allow users to filter their SOs by the old `search` name and the new `discover session` name
+// if we define the display name as a combination of them: "discover session / search".
+export const SavedSearchTypeDisplayName = 'discover session / search';
```

Comment:
```suggestion
export const SavedSearchTypeDisplayName = 'Discover session (formerly "search")';
```
Should we capitalize "Discover" if it's displayed to users? It might also be good to clearly indicate that "search" is the old name if filtering will still work.

Reply:
It's displayed to users on Saved Objects page.

With the suggested `Discover session (formerly "search")`:
<img width="1478" alt="Screenshot 2024-12-16 at 14 54 32" src="https://github.com/user-attachments/assets/33836605-7ae6-445b-affa-c11de4380fdc" />

Currently with `discover session / search` in the PR:
<img width="1477" alt="Screenshot 2024-12-16 at 14 56 13" src="https://github.com/user-attachments/assets/83b29d3e-f003-43c9-887f-629d45ef095e" />

We could also simplify and use `discover session`:
<img width="1476" alt="Screenshot 2024-12-16 at 14 57 39" src="https://github.com/user-attachments/assets/abd57b2c-954a-4c64-a673-3dc33db628b5" />

@kertal How should we define it?

> Should we capitalize "Discover" if it's displayed to users? 
It's not capitalize for other types, so probably not.

Reply:
For me `discover session` looks good, it's for advanced users, let's keep it consistent for now (short, lowercase). 

we can rebrand to 'data view (formerly "index pattern")' later on :)

Reply:
for context, I think it's helpful to share this information, because it saves admins / devs maybe a moment of confusion, but maybe it should be solved for all relabeled saved objects. but for now, let's keep this out of scope

Reply:
Thanks! Changed to `discover session` via https://github.com/elastic/kibana/pull/202217/commits/b0007795c4682399c91a888f91a67e7518cebb68

Reply:
thx, resolving